device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -216,6 +216,7 @@ public void unregisterMultiplexedDevice(ClientConfiguration config, boolean will
         }
 
         clientConfigurations.remove(config);
+        deviceSessionsOpenedLatches.remove(config.getDeviceId());
```

Comment:
Like the above, this state map was holding onto state that is no longer relevant once a device is unregistered.